File: backend/src/main/java/ca/metricalsky/winston/entity/view/AuthorDetailsView.java
package ca.metricalsky.winston.entity.view;

public interface AuthorDetailsView {

    String getAuthorId();

    Long getChannelCount();

    Long getVideoCount();

    Long getTotalCommentCount();

    Long getReplyCount();

    default Long getCommentCount() {
        return getTotalCommentCount() - getReplyCount();
    }
}

File: database/src/main/kotlin/ca/metricalsky/winston/database/view/AuthorDetailsView.kt
package ca.metricalsky.winston.database.view

interface AuthorDetailsView {

    fun getAuthorId(): String

    fun getChannelCount(): Long

    fun getVideoCount(): Long

    fun getTotalCommentCount(): Long

    fun getReplyCount(): Long

    fun getCommentCount(): Long {
        return getTotalCommentCount() - getReplyCount()
    }
}
